src/qt/nametablemodel.cpp

```diff
@@ -0,0 +1,658 @@
+        Qt::AlignLeft|Qt::AlignVCenter,     // Value
+        Qt::AlignRight|Qt::AlignVCenter,    // Expires in
+        Qt::AlignRight|Qt::AlignVCenter,    // Name Sataus
+    };
+}
+
+struct NameTableEntryLessThan
+{
+    bool operator()(const NameTableEntry &a, const NameTableEntry &b) const
+    {
+        return a.name < b.name;
+    }
+    bool operator()(const NameTableEntry &a, const QString &b) const
+    {
+        return a.name < b;
+    }
+    bool operator()(const QString &a, const NameTableEntry &b) const
+    {
+        return a < b.name;
+    }
+};
+
+// Returns true if new height is better
+bool NameTableEntry::CompareHeight(int nOldHeight, int nNewHeight)
+{
+    if(nOldHeight == NAME_NON_EXISTING)
+        return true;
+
+    // We use optimistic way, assuming that unconfirmed transaction will eventually become confirmed,
+    // so we update the name in the table immediately. Ideally we need a separate way of displaying
+    // unconfirmed names (e.g. grayed out)
+    if(nNewHeight == NAME_UNCONFIRMED)
+        return true;
+
+    // Here we rely on the fact that dummy height values are always negative
+    return nNewHeight > nOldHeight;
+}
+
+// Private implementation
+class NameTablePriv
+{
+public:
+    CWallet *wallet;
+    QList<NameTableEntry> cachedNameTable;
+    NameTableModel *parent;
+
+    NameTablePriv(CWallet *wallet, NameTableModel *parent):
+        wallet(wallet), parent(parent) {}
+
+    void refreshNameTable()
+    {
+        qDebug() << "NameTableModel::refreshNameTable";
+        cachedNameTable.clear();
+        std::map< std::string, NameTableEntry > vNamesO;
+
+        // unconfirmed (name_pending) names
+        JSONRPCRequest namePendingRequest;
+        namePendingRequest.strMethod = "name_pending";
+        namePendingRequest.params = NullUniValue;
+        namePendingRequest.fHelp = false;
+        UniValue pendingNames;
+
+        try {
+            pendingNames = tableRPC.execute(namePendingRequest);
+        } catch (const UniValue& e) {
+            // although we shouldn't typically encounter error here, we
+            // should continue and try to add confirmed names and
+            // pending names. show error to user in case something
+            // actually went wrong so they can potentially recover
+            UniValue message = find_value( e, "message");
+            LogPrintf ("name_pending lookup error: %s\n", message.get_str().c_str());
+        }
+
+        if(pendingNames.isArray())
+        {
+            for (const auto& v : pendingNames.getValues())
+            {
+                std::string name = find_value ( v, "name").get_str();
+                std::string data = find_value ( v, "value").get_str();
+                vNamesO[name] = NameTableEntry(
+                    name, data,
+                    NameTableEntry::NAME_UNCONFIRMED,
+                    "pending registration");
+                LogPrintf("found pending name: name=%s\n", name.c_str());
```

Comment:
I haven't explicitly tested this, but based on code review I suspect that this code will list all `name_anyupdate` transactions in the mempool, not just `name_anyupdate` transactions that belong to the user.  Checking the `"ismine"` field for each transaction returned by `name_pending` would fix that.  This is probably related to [the behavior reported by exmachinalibertas](https://www.reddit.com/r/Namecoin/comments/80hexm/help_me_track_down_a_possible_virus_related_to/).